be2-scala/src/main/scala/ch/epfl/pop/storage/DiskStorage.scala

```diff
@@ -0,0 +1,62 @@
+package ch.epfl.pop.storage
+
+import java.io.{File, IOException}
+import java.nio.charset.StandardCharsets
+
+import org.iq80.leveldb.{DB, Options, WriteBatch}
+import org.iq80.leveldb.impl.Iq80DBFactory.factory
+import org.slf4j.{Logger, LoggerFactory}
+
+class DiskStorage(val databaseFolder: String = DiskStorage.DATABASE_FOLDER) {
+
+  // create a logger for the database system
+  val logger: Logger = LoggerFactory.getLogger("DbLogger")
+
+  val options: Options = new Options()
+  options.createIfMissing(true)
+  options.cacheSize(DiskStorage.CACHE_SIZE)
+
+  val db: DB = {
+    try factory.open(new File(databaseFolder), options)
+    catch {
+      case e: IOException =>
+        logger.error("Could not open database folder '{}': {}", databaseFolder, e.getMessage)
+        throw e
+    }
+  }
+
+  def close(): Unit = db.close()
+
+
+  @throws [org.iq80.leveldb.DBException]
+  def read(key: String): String = {
+    new String(db.get(key.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8)
```

Comment:
I think it wouldn't hurt readability to split this in more lines , to avoid 3 nested function calls😉

Reply:
(Already corrected in #809 :) )